### PR TITLE
[BugFix] Fix aggregate with case-when mv bug

### LINKED ISSUE #15932: query with materialized view return error result
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
CREATE TABLE test0 (
    k1 INT NOT NULL COMMENT "k1",
    k2 INT NOT NULL COMMENT "k2",
    v1 INT COMMENT "v1"
)
DUPLICATE KEY(k1, k2)
DISTRIBUTED BY HASH(k1) BUCKETS 8;
```
2. INSERT INTO '....'
```
insert into test0 VALUES
(1,2,2),
(1,2,1),
(2,1,1),
(3,1,1)
```
3. CREATE MATERIALIZED VIEW
```
CREATE MATERIALIZED VIEW mv as
SELECT
k1,k2,sum(v1)
from test0
GROUP BY k1,k2;
```

### Expected behavior (Required)

```
MySQL [dps_data_center_stream]> select k1,k2,sum(case k1 when 1 then 1 end) from test0 group by k1,k2;
+------+------+--------------------------------+
| k1   | k2   | sum(CASE k1 WHEN 1 THEN 1 END) |
+------+------+--------------------------------+
|    1 |    2 |                              2 |
|    2 |    1 |                           NULL |
|    3 |    1 |                           NULL |
+------+------+--------------------------------+
3 rows in set (0.02 sec)
```

### Real behavior (Required)
```
MySQL [dps_data_center_stream]> select k1,k2,sum(case k1 when 1 then 1 end) from test0 group by k1,k2;
+------+------+--------------------------------+
| k1   | k2   | sum(CASE k1 WHEN 1 THEN 1 END) |
+------+------+--------------------------------+
|    3 |    1 |                           NULL |
|    2 |    1 |                           NULL |
|    1 |    2 |                              1 |
+------+------+--------------------------------+
3 rows in set (0.01 sec)
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.3.4

### PR DESCRIPTION
**Pull request by mchades**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #15932 

## Problem Summary(Required) ：

add ConstantRef check for Then-Clause in agg query  

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java|2|2|100.00%|[]|

**Comment by mchades** [comment]
run starrocks_admit_test

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java [hunk 1]
```diff
@@ -91,7 +91,8 @@ public Boolean visitCaseWhenOperator(CaseWhenOperator operator, Void context) {
                 return false;
             }
             ScalarOperator thenClause = operator.getThenClause(i);
-            if (!Utils.extractColumnRef(thenClause).stream().allMatch(aggregateColumns::contains)) {
+            if (!Utils.extractColumnRef(thenClause).stream().allMatch(aggregateColumns::contains) ||
+                    (thenClause.isConstantRef() && !((ConstantOperator) thenClause).isNull())) {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java:95]: I think may you need add thenClause.isConstant() check, like
select k1, sum(case k1 when 1 then abs(1) end )...

**Review by ABingHuang — DISMISSED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java:95]: @ABingHuang

Comment (mchades) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java:95]: `!((ConstantOperator) thenClause).isNull())` already include `thenClause.isConstant()`？ @Seaven

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java:95]: no, `ConstantOperator` only a literal, like '1'/'a'/NULL, `isConstant` means maybe an expression, like `abs(1)`, `1 + 1`, `case 1=abs(2) then 2`, all his inputs are `ConstantOperator`, but himselft not a `ConstantOperator`. we can compute some constant expression result in optimizer, but not all

Comment (mchades) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java]: thx for clear explanation，I have added henClause.isConstant() check. @Seaven

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/IsNoCallChildrenValidator.java|3|3|100.00%|[]|

**Comment by mchades** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**